File: LeetCode/JavaScript/#219 Contains Duplicate II.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
// My Way
var containsNearbyDuplicate = function(nums, k) {
    if(Array.from(new Set(nums)).toString()===nums.toString()) return false;
    var len = nums.length;
    for(let i=0;i<len;i++){
      for(let j = i+1;j<len;j++){
        if(nums[i]===nums[j]&&Math.abs(i-j)<=k){
          return true;
        }
      }
    }
    return false;  
};

//  Others Great way
var containsNearbyDuplicate = function(nums, k) {
    const dic = new Map();
    for (let i = 0; i < nums.length; i++) {
        // delete entries more than `k` away as we go
        if (i > k) {
            dic.delete(nums[i - 1 - k]);
        }
        let key = nums[i];
        // check if value was previously added
        if (dic.has(key)) {
            return true;
        }
        // set value: index
        dic.set(key, i);
    }
    return false;
};

File: LeetCode/Python/#219 Contains Duplicate II.py
#Beat 100%
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        if len(list(set(nums))) == len(nums):return False
        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                if nums[i]==nums[j] and abs(i-j)<=k:
                    return True
                if abs(i-j)>k:
                    j = len(nums)

        return False